app/cdap/components/hydrator/components/TopPanel/NameAndDescription/styles.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { Button, TextField, Tooltip, withStyles } from '@material-ui/core';
+import IconSVG from 'components/shared/IconSVG';
+import styled from 'styled-components';
+const colors = require('styles/colors.scss');
+const errorRed = colors.red03;
+const blueType = '#397cf1';
+
+export const EllipsisText = styled.div`
+  text-overflow: ellipsis;
+  white-space: nowrap;
+  overflow: hidden;
+`;
+
+export const PipelineName = styled(EllipsisText)`
+  font-size: 14px;
+  margin-top: 5px;
+  line-height: 17px;
+
+  ${({ errorName }) => errorName && `color: ${errorRed};`}
+`;
+
+export const PipelineDescription = styled(EllipsisText)`
+  color: gray;
+  font-size: 12px;
+`;
+
+export const CustomTooltip = withStyles(() => {
+  return {
+    tooltip: {
+      fontSize: '13px',
+    },
+  };
+})(Tooltip);
+
+export const ErrorTooltip = withStyles(() => {
+  return {
+    arrow: {
+      color: errorRed,
+    },
+    tooltip: {
+      background: errorRed,
+    },
+  };
+})(CustomTooltip);
+
+export const SaveButton = styled(Button)`
+  background-color: ${colors.blueprimary};
+  margin-right: 10px;
```

Comment:
Something to discuss: When I need to add space around a button or input element, I wrap it in a `span` or `div` and apply the margin/padding there. What do folks think about that approach vs styling the element directly?

This does not need to be changed before merging - just wanted to find out other peoples' thoughts.

Reply:
I've actually seen implementation of both. Personally I'd like to treat the two buttons equally so I'd like to either style the button directly, or style the wrapper to make it display flex and justify "space-between" content